ykrt/src/compile/mod.rs

```diff
@@ -20,13 +22,8 @@ pub(crate) mod jitc_llvm;
 pub mod jitc_yk;
 
 /// A failure to compile a trace.
-#[derive(Debug, thiserror::Error)]
```

Comment:
So this error type is now always a recoverable error? If so, shall we say that in a comment?

Reply:
There's no concept of "recoverable" or "unrecoverable" now: we just couldn't compile this particular trace. What the caller does with the failure is up to them. [In practise, mt will/should retry tracing the location a few times, and then eventually give up. It doesn't actually get that right now, though I have a half-finished branch which fixes that.]

Reply:
Got it. Conceptually recoverable.